ggml.c

```diff
@@ -2282,24 +2284,26 @@ static void ggml_vec_dot_q4_1(const int n, float * restrict s, const void * rest
         const float d0 = x[i].d;
         const float d1 = y[i].d;
 
-        const float m0 = x[i].m;
-        const float m1 = y[i].m;
+        const float m0 = x[i].m / d0;
+        const float m1 = y[i].m / d1;
```

Comment:
This change need more testing as scale smaller may lose some bits of m0/m1. I suggest you only change q4_0 which is current we are using.

Reply:
Yes, let's keep only the `Q4_0` change and merge